### PR TITLE
Split internal plugin code into multiple files

### PR DESCRIPTION
**Pull request by sashayakovtseva**
## Description of the Pull Request (PR):

Cleanup internal package for readability. Split code into multiple files so that `binary.go` is not 600+ lines.

#### Before submitting a PR, make sure you have done the following:

- Read the [Guidelines for Contributing](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](https://github.com/sylabs/singularity/blob/master/CHANGELOG.md) if necessary according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added tests to validate this PR and tested this PR locally with a `make testall`
- Based this PR against the appropriate branch according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](https://github.com/sylabs/singularity/blob/master/CONTRIBUTORS.md)

Attn: @singularity-maintainers

### CODE REVIEW

File: internal/pkg/plugin/meta.go [hunk 1]
```diff
@@ -0,0 +1,277 @@
+package plugin
```
Comment (WestleyK) [inline internal/pkg/plugin/meta.go:6]: Need a copyright header?

```suggestion
// Copyright (c) 2019, Sylabs Inc. All rights reserved.
// This software is licensed under a 3-clause BSD license. Please consult the
// LICENSE.md file distributed with the sources of this project regarding your
// rights to use or distribute this software.

package plugin
```

File: internal/pkg/plugin/sif.go [hunk 2]
```diff
@@ -0,0 +1,146 @@
+package plugin
```
Comment (WestleyK) [inline internal/pkg/plugin/sif.go:6]: Same here

```suggestion
// Copyright (c) 2019, Sylabs Inc. All rights reserved.
// This software is licensed under a 3-clause BSD license. Please consult the
// LICENSE.md file distributed with the sources of this project regarding your
// rights to use or distribute this software.

package plugin
```

File: internal/pkg/plugin/sif.go [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+import (
+	"encoding/json"
+
+	"github.com/sylabs/singularity/internal/pkg/sylog"
+
```
Comment (WestleyK) [inline internal/pkg/plugin/sif.go]: Extra newline

```suggestion
```

**Review by WestleyK — CHANGES_REQUESTED** [review]
>**WestleyK** on behalf of **WestleyK**, requested changes on this PR

Couple suggestions, otherwise looks good to me!

File: internal/pkg/plugin/sif.go (hunk 3 above)
Comment (dtrudg) [reply internal/pkg/plugin/sif.go]: With the newline removed sort order should have sif before sylog

**Comment by dtrudg** [comment]
Hi @sashayakovtseva - if you can enable edits from maintainers on hopefully non-controversial things like this we can add commit minor things e.g. the copyright headers to get a faster turnaround on merging. Cheers.

File: internal/pkg/plugin/sif.go (hunk 3 above)
Comment (sashayakovtseva) [reply internal/pkg/plugin/sif.go]: Thanks @WestleyK!

**Comment by sashayakovtseva** [comment]
@dctrud Github says edits were enabled

**Comment by dtrudg** [comment]
> @dctrud Github says edits were enabled

Strange - I tried to commit the copyright changes, but it said I didn't have permission to push to the branch. I'll try to work out why.

**Review by WestleyK — APPROVED** [review]
Looks good to me, thanks!

**Review by tri-adam — APPROVED** [review]
LGTM

**Pull request merged**